### ISSUE TITLE
Feature: Add more data type support for key

### ISSUE THREAD

**Issue by jaredzhou** [ConduitIO/conduit-connector-postgres#185]
### Feature description

this pg connector only support int type key for table for now , i don't see the reason, i think other type like varchar is possible, and it's common to use uuid or varchar  as primary key

**Comment by lovromazgon** [comment]
We currently don't support them because we were focusing on making it possible to do an incremental snapshot (i.e. if the snapshot is interrupted we don't have to start the snapshot again from scratch). This is specifically important for huge tables where the snapshot can take hours or even days. That's why we currently have the restriction for keys to be an ordinal type that's auto-incremented. However, we do plan to add support for other types, with the caveat of not being able to do an incremental snapshot.

**Comment by jaredzhou** [comment]
that make sense, glad to see the support for varchar.

**Comment by lovromazgon** [comment]
I realized that you can still use the connector to move data continuously using logical replication (CDC), the restriction on the primary key type only applies to snapshots.

**Comment by rauanmayemir** [comment]
This makes sense. I wish UUID was also supported with the expectation that user must use lexicographically sortable version (v6/v7).

**Comment by lyuboxa** [comment]
> This makes sense. I wish UUID was also supported with the expectation that user must use lexicographically sortable version (v6/v7).

We can relax this v6+, it will be harder for the connector to determine if the UUID type's of v6+, so it will need to come with a config and allow the user to decide which will override the key detection.

**Comment by rauanmayemir** [comment]
Yeah, I use ULID which is sortable and similar to V6, but not quite V6.

**Comment by briankski** [comment]
+1 for this. Really was hoping to get off of debezium, but most of our tables use a UUID as a primary key
